app/controllers/accounts_controller.rb

```diff
@@ -15,6 +15,7 @@ def show
       sp_name: decorated_session.sp_name,
       decorated_user: current_user.decorate,
       locked_for_session: pii_locked_for_session?(current_user),
+      account_reset_requested: current_user&.account_reset_request,
```

Comment:
I mean it may be necessary to use `identity-idp/app/services/account_reset/find_pending_request_for_user.rb` (`AccountReset::FindPendingRequestForUser`), specifically to search for `AccountResetRequests` with the condition `cancelled_at: nil`.

Reply:
ahhhh yes, maybe it would help to have a different associaiton, such as a `user.pending_account_reset_request` but also now that I look at this, the result we care about here is if it exists or not... not the whole record so we could simplify like this (incorporating the `cancelled_at: nil` logic would be different potentially)

```suggestion
      account_reset_requested: current_user&.account_reset_request&.exist?,
```

Reply:
I believe you still want to filter the `AccountResetRequests` by the property `cancelled_at`.

If you want to show the message that their account is not being deleted, search for the presence of a user's `AccountResetRequest` where `cancelled_at` is not `nil`. Otherwise, if you're showing a message warning that their account will be deleted, you want to filter for the presence of an `AccountResetRequest` where `cancelled_at` is `nil`.

For showing a message that the deletion has been cancelled, you could check that the value of `cancelled_at` is within the past 24 hours.